pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -690,3 +692,16 @@ func (r *Builder) host(hostID string) (host *model.Host, err error) {
 func trimBackingFileName(fileName string) string {
 	return backingFilePattern.ReplaceAllString(fileName, ".vmdk")
 }
+
+// Build an OvirtVolumePopulator for XDiskAttachment and source URL, not supported for vSphere
+func (r *Builder) OvirtVolumePopulator(da ovirt.XDiskAttachment, sourceUrl *url.URL) *api.OvirtVolumePopulator {
```

Comment:
We need to rethink this at some point, it's way too specific, and it forces ovirt imports in a vsphere package

Reply:
do you mean renaming it?

Reply:
not the name as we need the type here too, but perhaps it would make more sense to move it out to be a standalone function and not part of the builder interface, it doesn't depend on too many fields from `builder`